instrumentation/mongo/mongo-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/mongo/v3_1/MongoClientInstrumentationModule.java

```diff
@@ -67,12 +66,7 @@ public MongoClientInstrumentationModule() {
     public static void injectTraceListener(
         @Advice.This MongoClientOptions.Builder builder,
         @Advice.FieldValue("commandListeners") List<CommandListener> commandListeners) {
-      for (CommandListener commandListener : commandListeners) {
-        if (commandListener instanceof TracingCommandListener) {
-          return;
-        }
-      }
```

Comment:
this may have been protecting against calling `build()` twice on the same builder

Reply:
Ah good point, restored dedupe. Strangely a new test I added doesn't fail even without these though, haven't dug into why that might be.

Reply:
> a new test I added doesn't fail even without these though

oh no, maybe it's not needed after all(?)

Reply:
I wonder if our workForTraces doesn't actually check for extra traces? I thought it did

Reply:
try putting a sleep before it to give the second trace time to get there

Reply:
There isn't really anywhere to put a sleep - build just causes two listeners to be registered, which I could confirm in the debugger and Thread.dumpStack. Weird but not sure I'll dig too much into it